driver-core/src/test/java/com/datastax/driver/core/policies/ErrorAwarePolicyTest.java

```diff
@@ -0,0 +1,388 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core.policies;
+
+import com.codahale.metrics.Clock;
+import com.codahale.metrics.Meter;
+import com.datastax.driver.core.*;
+import com.datastax.driver.core.exceptions.*;
+import com.google.common.util.concurrent.Uninterruptibles;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.scassandra.http.client.Result;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.datastax.driver.core.TestUtils.nonQuietClusterCloseOptions;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.scassandra.http.client.PrimingRequest.queryBuilder;
+import static org.scassandra.http.client.PrimingRequest.then;
+import static org.scassandra.http.client.Result.*;
+
+public class ErrorAwarePolicyTest {
+
+    private QueryTracker queryTracker;
+    private Clock clock;
+    private ScassandraCluster sCluster;
+    private long currentTick;
+
+    @BeforeMethod(groups = "short")
+    public void setUp() {
+        queryTracker = new QueryTracker();
+        // each execute will elapse the system clock 1 second, 2 seconds in the case where the error threshold is hit.
+        clock = mockClock(1000, TimeUnit.MILLISECONDS);
+        currentTick = 0L;
+        sCluster = ScassandraCluster.builder().withNodes(2).build();
+        sCluster.init();
+
+        // By default node 1 should always fail, 2 succeed.
+        prime(1, unauthorized);
+        prime(2, success);
+    }
+
+    @AfterMethod(groups = "short")
+    public void tearDown() {
+        sCluster.stop();
+    }
+
+    private Cluster.Builder builder(LoadBalancingPolicy lbp) {
+        return Cluster.builder()
+                .withNettyOptions(nonQuietClusterCloseOptions)
+                .addContactPoints(sCluster.address(1).getAddress())
+                .withPort(sCluster.getBinaryPort())
+                .withLoadBalancingPolicy(lbp);
+    }
+
+    /**
```

Comment:
This documentation I wrote here makes the test methodology seem way more complicated then it actually is.  Originally the Mock clock would increment `getTick()` whenever it was invoked.  

Instead I changed it so it's up to the user to call `advanceTick()` whenever they want the `Meter` to tick when the next `mark` is called (i.e. when an error happens).  This allows fine grained control of when m1rate is calculated, which allows us a straightforward way to get the Meter in a desired state that exercises the policy.

As far as System time goes, I made it so whenever `Clock.getTime()` is called it returns 1 second more than the previous call.  This creates a straightforward side effect that whenever a query is executed, 1 second has elapsed since the last query.  This ties the behavior of the test to that side effect, which may not be that great, but I figured it was easier than advancing the system timer by hand too like is done for `advanceTick`, although it may be smart to control that by hand as well, what do you think?